pkg/apis/core/validation/validation.go

```diff
@@ -5579,6 +5579,13 @@ func ValidateNamespace(namespace *core.Namespace) field.ErrorList {
 	for i := range namespace.Spec.Finalizers {
 		allErrs = append(allErrs, validateFinalizerName(string(namespace.Spec.Finalizers[i]), field.NewPath("spec", "finalizers"))...)
 	}
+	if utilfeature.DefaultFeatureGate.Enabled(features.DefaultNamespaceLabelStamping) {
+		if namespace.Labels["kubernetes.io/metadata.name"] != namespace.Name {
+			allErrs = append(allErrs, field.Invalid(field.NewPath("metadata", "labels").Key("kubernetes.io/metadata.name"), namespace.Labels["kubernetes.io/metadata.name"], fmt.Sprintf("must be %s", namespace.Name))
+		}
+	} else {
+		klog.V(4).Infof("Automatic namespace stamping is disabled. not labelling %v", &namespace.ObjectMeta.Name)
```

Comment:
this will run always that the feature is disabled, @liggitt does you comment in defaults to avoid logging apply here too?
https://github.com/kubernetes/kubernetes/pull/96968#discussion_r537809260

Reply:
Good question , also @liggitt  if we *do* decide to print this should it be a `klog.Warn` ?

Reply:
yeah, I would not expect us to log from validation like this... if the feature is off, we shouldn't flood the apiserver log with messages on every namespace write.

Reply:
ok i will look at  the implementation implied in the original kep
```
we can send them from strategy methods (PrepareForCreate / PrepareForUpdate / Validate / ValidateUpdate) and anything we plumb the request context to
```

Reply:
if we stomp in defaulting, I don't think we can send the warnings we want in Create/Update-specific paths

Reply:
ok, ill update the KEP accordingly

Reply:
Updated the KEP text to mention that this specific implementation 
wont satisfy `warnings`.

I think thats something we need to define as a requirement, or not, explicitly ....